app/scripts/controllers/serviceInstance.js

```diff
@@ -37,25 +58,68 @@ angular.module('openshiftConsole')
       });
     };
 
-    var updateServiceClass = function() {
-      if ($scope.serviceClass) {
-        return;
+    var updateParameterData = function() {
+      if (secretWatcher) {
+        DataService.unwatch(secretWatcher);
+        secretWatcher = undefined;
       }
 
-      ServiceInstancesService.fetchServiceClassForInstance($scope.serviceInstance).then(function(serviceClass) {
-        $scope.serviceClass = serviceClass;
-        $scope.displayName = serviceInstanceDisplayName($scope.serviceInstance, serviceClass);
-        updateBreadcrumbs();
+      $scope.parameterData = {};
+      _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function(key) {
+        $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
+      });
+
+      $scope.parameterData = angular.extend($scope.parameterData, _.get($scope.serviceInstance, 'spec.parameters', {}));
+
+      _.each(_.get($scope.serviceInstance, 'spec.parametersFrom'), function(parametersSource) {
+        secretWatcher = DataService.watchObject("secrets", _.get(parametersSource, 'secretKeyRef.name'), $scope.projectContext, function (secret) {
```

Comment:
If there's more than one secret, you'll lose the previous `secretWatcher` reference and leak watches